### PR TITLE
DM-20551: Add a number of bug-fixes and requested features to display_matplotlib

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**
Support sexagesimal formatting of coordinates
   Either via a parameter to the ctor, or via an extension API;
   both called useSesagesimal

   Also make dpi a real arg, not kwargs (why did I do that?)
Fix matplotlib errors/warnings
Show WCS in degrees not radians
Set better vmax for asinh stretches
Add support for passing a matplotlib figure as the frame
set_colorbar() the default
Obey scale(..., maskedPixel=XXX)
   Currently only implemented for minmax; see DM-16565

### CODE REVIEW

File: python/lsst/display/matplotlib/matplotlib.py [hunk 1]
```diff
@@ -257,10 +364,24 @@ def format_coord(x, y, wcs=self._wcs, x0=self._xy0[0], y0=self._xy0[1],
...
+
+                    if _useSexagesimal[0]:
+                        from astropy import units as u
+                        from astropy.coordinates import Angle as apAngle
+
+                        kwargs = dict(sep=':', pad=True, precision=2)
+                        ra = apAngle(ra*u.deg).to_string(unit=u.hour, **kwargs)
+                        dec = apAngle(dec*u.deg).to_string(unit=u.deg, **kwargs)
+                    else:
+                        ra = ("%9.4f") % ra
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: I don't think the parens are doing anything here. You can alternatively do:
```python
ra = f"{ra:9.4f}"
```

File: python/lsst/display/matplotlib/matplotlib.py [hunk 2]
```diff
@@ -273,14 +394,13 @@ def format_coord(x, y, wcs=self._wcs, x0=self._xy0[0], y0=self._xy0[1],
...
-            if isinstance(a, AxesImage):
-                a.get_cursor_data = lambda ev: None  # disabled
-
-        self._figure.tight_layout()
+        # Stop images from reporting their value as we've already
+        # printed it nicely
+        for a in ax.get_images():
+            a.get_cursor_data = lambda ev: None  # disabled
+
+        if False:
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: If we are keeping this block in but disabled may be explain in what context we should use `True` here?

File: python/lsst/display/matplotlib/matplotlib.py [hunk 3]
```diff
@@ -600,6 +734,9 @@ def _pan(self, colc, rowc):
     def _getEvent(self, timeout=-1):
         """Listen for a key press, returning (key, x, y)"""
 
+        if timeout < 0:
+            timeout = 24*3600           # -1 generates complaints in QTimer::singleShot
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: Do we want to comment why 1 day is a good timeout number?

File: python/lsst/display/matplotlib/matplotlib.py [hunk 4]
```diff
@@ -617,9 +754,9 @@ class BlockingKeyInput(BlockingInput):
     Callable class to retrieve a single keyboard click
     """
     def __init__(self, fig):
-        r"""Create a BlockingKeyInput
+        """Create a BlockingKeyInput
 
-        \param fig The figure to monitor for keyboard events
+        @param fig The figure to monitor for keyboard events
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: Maybe switch to numpydoc like some of the changes earlier?

File: python/lsst/display/matplotlib/matplotlib.py [hunk 5]
```diff
@@ -679,13 +816,30 @@ def __init__(self, minimum=0, dataRange=1, Q=8):
...
         """
-        Normalize.__init__(self)
-
         # The object used to perform the desired mapping
         self.mapping = afwRgb.AsinhMapping(minimum, dataRange, Q)
 
+        vmin, vmax = self._getMinMaxQ()[0:2]
+        if vmax*Q > vmin:
+            vmax *= Q
+        Normalize.__init__(self, vmin, vmax)
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: Use `super().__init__(vmin, vmax)`

File: python/lsst/display/matplotlib/matplotlib.py [hunk 6]
```diff
@@ -696,11 +850,13 @@ def __init__(self, image=None, Q=8):
 
         See Lupton et al., PASP 116, 133
         """
-        Normalize.__init__(self)
 
         # The object used to perform the desired mapping
         self.mapping = afwRgb.AsinhZScaleMapping(image, Q)
 
+        vmin, vmax = self._getMinMaxQ()[0:2]
+        Normalize.__init__(self, vmin, vmax)
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: Add a comment if you are deliberately bypassing the `AsinhNormalize.__init__` to explain why `super()` is not being used.

File: python/lsst/display/matplotlib/matplotlib.py [hunk 7]
```diff
@@ -713,11 +869,11 @@ def __init__(self, image=None, nSamples=1000, contrast=0.25):
                         median (default: 0.25)
         """
 
-        Normalize.__init__(self)
-
         # The object used to perform the desired mapping
         self.mapping = afwRgb.ZScaleMapping(image, nSamples, contrast)
 
+        Normalize.__init__(self, self.mapping.minimum[0], self.mapping.maximum)
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: `super()` ?

File: python/lsst/display/matplotlib/matplotlib.py [hunk 8]
```diff
@@ -727,8 +883,7 @@ def __init__(self, minimum=0, maximum=1):
         @param minimum  Minimum value to display
         @param maximum  Maximum value to display
         """
-
-        Normalize.__init__(self)
-
         # The object used to perform the desired mapping
         self.mapping = afwRgb.LinearMapping(minimum, maximum)
+
+        Normalize.__init__(self, self.mapping.minimum[0], self.mapping.maximum)
```
Comment (timj) [inline python/lsst/display/matplotlib/matplotlib.py]: `super()`?

File: examples/display_matplotlib.ipynb [hunk 9]
```diff
@@ -227,12 +226,47 @@
...
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "im = afwImage.MaskedImageF(100, 50)\n",
+    "afwMath.randomGaussianImage(im.image, afwMath.Random())\n",
+    "\n",
+    "sim = im[afwGeom.BoxI(afwGeom.PointI(0, 0), afwGeom.ExtentI(im.getWidth()//2, im.getHeight()))]\n",
```
[the remark this answers is not in the document]
Comment (RobertLuptonTheGood) [reply examples/display_matplotlib.ipynb]: I did that; maybe it was lost when I rebased to master and picked up your flake8 changes?

File: python/lsst/display/matplotlib/matplotlib.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply python/lsst/display/matplotlib/matplotlib.py]: I changed the `if False` block to a comment on why I didn't use tight_layout()

File: python/lsst/display/matplotlib/matplotlib.py (hunk 3 above)
Comment (RobertLuptonTheGood) [reply python/lsst/display/matplotlib/matplotlib.py]: It just needs to be a long time;  I added that

File: python/lsst/display/matplotlib/matplotlib.py (hunk 4 above)
Comment (RobertLuptonTheGood) [reply python/lsst/display/matplotlib/matplotlib.py]: This is an orthogonal change that I'd rather do when I'm not rushing to get this into a weekly

File: python/lsst/display/matplotlib/matplotlib.py (hunk 5 above)
Comment (RobertLuptonTheGood) [reply python/lsst/display/matplotlib/matplotlib.py]: Python2 code.  I'll make all the super changes requested

**Pull request merged**